rs/nervous_system/integration_tests/tests/set_following.rs

```diff
@@ -0,0 +1,516 @@
+
+    // Get an ID of an SNS neuron that can submit proposals. We rely on the fact that this
+    // neuron either holds the majority of the voting power or the follow graph is set up
+    // s.t. when this neuron submits a proposal, that proposal gets through without the need
+    // for any voting.
+    let (my_sns_neuron_id, sender) = sns::governance::find_neuron_with_majority_voting_power(
+        &pocket_ic,
+        sns.governance.canister_id,
+    )
+    .await
+    .expect("cannot find SNS neuron with dissolve delay over 6 months.");
+
+    let another_sns_neuron_id = sns::governance::find_another_neuron(
+        &pocket_ic,
+        sns.governance.canister_id,
+        my_sns_neuron_id.clone(),
+    )
+    .await
+    .expect("cannot find a second SNS neuron.");
+
+    // Smoke test
+    assert_ne!(another_sns_neuron_id, my_sns_neuron_id);
+
+    let expected_followee = Followee {
+        neuron_id: Some(another_sns_neuron_id.clone()),
+        alias: Some("Bob Dylan".to_string()),
+    };
+
+    let pocket_ic_agent = PocketIcAgent {
+        pocket_ic: &pocket_ic,
+        sender: sender.into(),
+    };
+
+    let governance_canister = GovernanceCanister {
+        canister_id: sns.governance.canister_id,
+    };
+
+    let ManageNeuronResponse { command } = governance_canister
+        .follow(
+            &pocket_ic_agent,
+            my_sns_neuron_id.clone(),
+            Follow {
+                function_id: 0, // catch-all
+                followees: vec![another_sns_neuron_id.clone()],
+            },
+        )
+        .await
+        .unwrap();
+
+    assert_eq!(command, Some(Command::Follow(FollowResponse {})));
+
+    let ManageNeuronResponse { command } = governance_canister
+        .follow(
+            &pocket_ic_agent,
+            my_sns_neuron_id.clone(),
+            Follow {
+                function_id: 1, // E.g., MOTION (essentially, any non-critical proposal type)
+                followees: vec![another_sns_neuron_id.clone()],
+            },
+        )
+        .await
+        .unwrap();
+
+    assert_eq!(command, Some(Command::Follow(FollowResponse {})));
+
+    // Follow someone on 4 out of 5 non-critical topics and all critical topics.
+    // Topic::ApplicationBusinessLogic is not yet followed, as this topic contains the proposal
+    // type `MOTION` (ID 1), for which we have legacy following.
+    for topic in [
+        Topic::DaoCommunitySettings,
+        Topic::SnsFrameworkManagement,
+        Topic::DappCanisterManagement,
+        // `Topic::ApplicationBusinessLogic` is the only one missing from the list.
+        Topic::Governance,
+        Topic::TreasuryAssetManagement,
+        Topic::CriticalDappOperations,
+    ] {
+        let ManageNeuronResponse { command } = governance_canister
+            .set_following(
+                &pocket_ic_agent,
+                my_sns_neuron_id.clone(),
+                SetFollowing {
+                    topic_following: vec![FolloweesForTopic {
```

Comment:
Might be a good idea (for test time efficiency) to combine all the above topics into a single proposal?  Doing it this way means there are going to be more ticks and seconds in processing it, which seems noticeable in PocketIC tests.

Reply:
This was indeed intentional. I see your point about the test's runtime. Let me try to address this also in the follow-up where I move these tests.

Reply:
My plan is to reduce the number of commands to just 2 even in the incremental case, that should be enough to assert the property that incremental addition works as intended.